### PR TITLE
Initial addition of TefnutMain.py and sensors.py files

### LINKED ISSUE #4: Create 'read sensor' code

### PR DESCRIPTION
**Pull request by msjbak**
Will close #4

### CODE REVIEW

**Comment by bzillins** [comment]
Are you requesting me to do anything here?

**Comment by msjbak** [comment]
Review and provide feedback as commits are made

**Comment by msjbak** [comment]
Also added a cron job on the BBB to run Main every hour. We will get to see tomorrow if it worked. if it does, I will have a day of hourly data for humidity and moisture.

**Comment by msjbak** [comment]
Cron job didn't work :(

**Comment by notexpected** [comment]
Boo, what went wrong?

**Comment by msjbak** [comment]
dunno, added the crontab to the branch

**Comment by notexpected** [comment]
Did you try running the py script twice to ensure that data is not being overwritten?

**Comment by bzillins** [comment]
I would be surprised if that was an issue. Joanna is correctly opening the
output file with the "ab" flag in TefnutMain.py

From the docs:
The first argument is a string containing the filename. The second argument
is another string containing a few characters describing the way in which
the file will be used. mode can be 'r' when the file will only be read, 'w'
for only writing (an existing file with the same name will be erased), and
'a' opens the file for appending; any data written to the file is
automatically added to the end. 'r+' opens the file for both reading and
writing. The mode argument is optional; 'r' will be assumed if it’s omitted.

https://docs.python.org/2/tutorial/inputoutput.html#reading-and-writing-files

On Thu, Nov 19, 2015 at 10:54 PM, Steven Feldman notifications@github.com
wrote:

> Did you try running the py script twice to ensure that data is not being
> overwritten?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/msjbak/Tefnut/pull/7#issuecomment-158272672.

## 

Brenton Zillins
ME Student
754-273-6866

**Comment by bzillins** [comment]
My beat guesses:

Add this to the top of mainsensors.py
#!/usr/bin/python

Change to an absolute path in the crontab. It is likely the cron process is
not running as the same user that you are logged in to the terminal as. If
that is the case the cron job will be hitting a different absolute path due
to your relative referencing.

This seems to be a good debugging thread.

http://stackoverflow.com/questions/12534135/crontab-not-running-my-python-script
I would be surprised if that was an issue. Joanna is correctly opening the
output file with the "ab" flag in TefnutMain.py

From the docs:
The first argument is a string containing the filename. The second argument
is another string containing a few characters describing the way in which
the file will be used. mode can be 'r' when the file will only be read, 'w'
for only writing (an existing file with the same name will be erased), and
'a' opens the file for appending; any data written to the file is
automatically added to the end. 'r+' opens the file for both reading and
writing. The mode argument is optional; 'r' will be assumed if it’s omitted.

https://docs.python.org/2/tutorial/inputoutput.html#reading-and-writing-files

On Thu, Nov 19, 2015 at 10:54 PM, Steven Feldman notifications@github.com
wrote:

> Did you try running the py script twice to ensure that data is not being
> overwritten?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/msjbak/Tefnut/pull/7#issuecomment-158272672.

## 

Brenton Zillins
ME Student
754-273-6866

File: sensors.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            temperatureF = ((float(temp)*9)/5)+32
+        else:
+            temperatureC = 'not found'
+            temperatureF = 'not found'
+
+
+        if humid is not None:
+            humidity = humid
+	else:
+            humidity = 'not found'
```
Comment (notexpected) [inline sensors.py:24]: None type maybe a better choice than a string

File: sensors.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import Adafruit_DHT
+import Adafruit_BBIO.ADC as ADC
+
+
+class HumidTempData(object):
```
Comment (notexpected) [inline sensors.py:5]: You should add a description of return values here

**Pull request merged**